src/System.Management.Automation/engine/InternalCommands.cs

```diff
@@ -351,80 +386,107 @@ protected override void ProcessRecord()
+                                if (ShouldProcess(_targetString, propertyAction))
+                                {
+                                    WriteObject(member.Value);
                                 }
+                                return;
+                            }
 
-                                PSMethodInfo targetMethod = member as PSMethodInfo;
-                                Dbg.Assert(targetMethod != null, "targetMethod should not be null here.");
-                                try
-                                {
-                                    // should process
-                                    string methodAction = String.Format(CultureInfo.InvariantCulture,
-                                        InternalCommandStrings.ForEachObjectMethodActionWithoutArguments, targetMethod.Name);
+                            PSMethodInfo targetMethod = member as PSMethodInfo;
+                            Dbg.Assert(targetMethod != null, "targetMethod should not be null here.");
+                            try
+                            {
+                                // should process
+                                string methodAction = String.Format(CultureInfo.InvariantCulture,
+                                    InternalCommandStrings.ForEachObjectMethodActionWithoutArguments, targetMethod.Name);
 
-                                    if (ShouldProcess(_targetString, methodAction))
+                                if (ShouldProcess(_targetString, methodAction))
+                                {
+                                    if (!BlockMethodInLanguageMode(InputObject))
                                     {
-                                        if (!BlockMethodInLanguageMode(InputObject))
-                                        {
-                                            object result = targetMethod.Invoke(Utils.EmptyArray<object>());
-                                            WriteToPipelineWithUnrolling(result);
-                                        }
+                                        object result = targetMethod.Invoke(Utils.EmptyArray<object>());
+                                        WriteToPipelineWithUnrolling(result);
                                     }
                                 }
-                                catch (PipelineStoppedException)
+                            }
+                            catch (PipelineStoppedException)
+                            {
+                                // PipelineStoppedException can be caused by select-object
+                                throw;
+                            }
+                            catch (Exception ex)
+                            {
+                                MethodException mex = ex as MethodException;
+                                if (mex != null && mex.ErrorRecord != null && mex.ErrorRecord.FullyQualifiedErrorId == "MethodCountCouldNotFindBest")
```

Comment:
Please use C# 7 syntax
```c#
if ( ex is MethodException mex ...
```

Reply:
This code is also unrelated to my change (harder to tell for this one, but it's just moved a bit).